### PR TITLE
v0.23.0: cherry-pick docs updates from master

### PR DESCRIPTION
**Pull request by alindima**
## Reason for This PR

Cherry-picked relevant documentation updates for the v0.23.0 release.

## Description of Changes

- various doc updates, including: release pace, prod-host-setup, snapshot security & uniqueness

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -143,7 +143,21 @@ Firecracker microVMs expose access to a minimal MicroVM-Metadata Service
...
+
+##### Seccomp
+
+Seccomp filters are used by default to further limit the system calls Firecracker
+can use. There are 3 possible levels of seccomp filtering, configurable by passing
+a command line argument to Firecracker: 0 (disabled), 1 (whitelists a set of
+trusted system calls by their identifiers) and 2 (whitelists a set of trusted
+system calls with trusted parameter values), the latter being the most
+restrictive and the recommended one. The filters are loaded in the Firecracker
+process, immediately before the execution of the untrusted guest code starts.
```
Comment (sandreim) [inline docs/design.md:158]: This is a bit confusing, do we really trust any guest code ?
```suggestion
recommended and restricted one. Firecracker enables the filters on all threads 
before the execution of guest code starts.
```

Comment (alindima) [reply docs/design.md:158]: We don't trust any guest code. We just emphasise the fact that the guest code is untrusted.

Comment (alindima) [reply docs/design.md:158]: the phrasing above is actually the exact same phrasing used on master, even before the docs update commit. you think it makes sense to alter it here? My opinion is that we should open a PR on mainline about any issues regarding this phrasing

Comment (sandreim) [reply docs/design.md:158]: I think we can downgrade this as a nit.

**Review by sandreim — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Pull request merged**